x/ccv/provider/keeper/proposal.go

```diff
@@ -289,7 +289,23 @@ func (k Keeper) MakeConsumerGenesis(
 		bondedValidators = append(bondedValidators, val)
 	}
 
-	nextValidators := k.ComputeNextEpochConsumerValSet(ctx, chainID, bondedValidators)
+	var nextValidators []types.ConsumerValidator
+
+	considerOnlyOptIn := func(validator stakingtypes.Validator) bool {
+		consAddr, err := validator.GetConsAddr()
+		if err != nil {
+			return false
+		}
+		return k.IsOptedIn(ctx, chainID, types.NewProviderConsAddress(consAddr))
+	}
+
+	if topN, found := k.GetTopN(ctx, chainID); found {
```

Comment:
The `TopN` value for the consumer isn't set at this poin since SetTopN() is called [after](https://github.com/cosmos/interchain-security/blob/03af237c800211642e81759b5f2e23bdcbad1113/x/ccv/provider/keeper/proposal.go) `CreateConsumerClientInCachedCtx`. 
So `prop.Top_N` should be used here.

Reply:
Good catch! Fixed.